partiql-planner/src/main/kotlin/org/partiql/planner/transforms/RexConverter.kt

```diff
@@ -145,11 +148,28 @@ internal object RexConverter {
                 when (it) {
                     is Expr.Path.Step.Index -> {
                         val key = visitExprCoerce(it.key, context)
-                        rexOpPathStepIndex(key)
+                        when (val astKey = it.key) {
+                            is Expr.Lit -> when (astKey.value) {
+                                is StringValue -> rexOpPathStepKey(key)
+                                else -> rexOpPathStepIndex(key)
+                            }
+                            is Expr.Cast -> when (astKey.asType is Type.String) {
+                                true -> rexOpPathStepKey(key)
+                                false -> rexOpPathStepIndex(key)
+                            }
+                            else -> rexOpPathStepIndex(key)
+                        }
                     }
                     is Expr.Path.Step.Symbol -> {
-                        val identifier = AstToPlan.convert(it.symbol)
-                        rexOpPathStepSymbol(identifier)
+                        when (it.symbol.caseSensitivity) {
+                            org.partiql.ast.Identifier.CaseSensitivity.SENSITIVE -> rexOpPathStepKey(
+                                rex(
+                                    type = StaticType.STRING,
+                                    op = rexOpLit(stringValue(it.symbol.symbol))
+                                )
+                            )
```

Comment:
We want this to be a `rexOpPathSymbol` right? For global resolution.

```
a."x" != a["x"]  -- !! only for global variable names, the prior is treated as a qualified identifier
```

Reply:
With your recent PR for internalizing the unresolved IRs, I added the retention of this subtle difference to the internal IR. Then, converting from the internal to the public, I transform it accordingly. See the updated PR description.